### PR TITLE
Project list page updates

### PR DESCRIPTION
**Pull request by selfcontained**
## Description
This is a followup PR to #15316 to share some updates made there with extracting a remove project modal. A couple other related updates on the Project list page are included.

* Simplify project list by extracting a `<ProjectListItem/>` component. This let us drop a bunch of state management we were doing around which project to remove.
* Memoized the sorting/filtering of projects so it only occurs when a relevant value changes instead of every render.
* Added a flag to track projects having been loaded to avoid a content flash on initial page load

## How to test

Preview Env: https://bmh-projec70ae423ab3.preview.gitpod-dev.com/

* Navigate to your personal, or a team's Projects list.
* Try and Remove a project via the overflow menu action.
* It should open a confirmation modal like it used to (this is using a new component for that now).
* Canceling should close it, Removing should remove the project, and close the modal.

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/dashboard/src/projects/ProjectSettings.tsx [hunk 1]
```diff
@@ -261,12 +261,12 @@ export default function () {
                 )}
             </div>
             <div>
-                <h3 className="mt-12">Delete Project</h3>
+                <h3 className="mt-12">Remove Project</h3>
                 <p className="text-base text-gray-500 dark:text-gray-400 pb-4">
-                    Removing the project from this team will also remove team members access to it.
+                    Removing this project will also remove all associated data with this project, including workspaces.
```
Comment (svenefftinge) [inline components/dashboard/src/projects/ProjectSettings.tsx:266]: Sorry, I might have confused you with a previous comment. We do not remove (delete) any associated data atm. I was just wondering if we should and/or what the consequences of deleting a project are.

File: components/dashboard/src/projects/ProjectSettings.tsx [hunk 2]
```diff
@@ -261,12 +261,12 @@ export default function () {
                 )}
             </div>
             <div>
-                <h3 className="mt-12">Delete Project</h3>
+                <h3 className="mt-12">Remove Project</h3>
```
Comment (svenefftinge) [inline components/dashboard/src/projects/ProjectSettings.tsx:264]: We are actually deleting the project here. Remove suggests that it is just somehow disconnected.

**Review by svenefftinge — APPROVED** [review]
LGTM 

/hold because of the two comments

File: components/dashboard/src/projects/ProjectSettings.tsx (hunk 2 above)
Comment (svenefftinge) [reply components/dashboard/src/projects/ProjectSettings.tsx:264]: cc @gtsiolis

File: components/dashboard/src/projects/ProjectSettings.tsx (hunk 1 above)
Comment (gtsiolis) [inline components/dashboard/src/projects/ProjectSettings.tsx:266]: **thought**: While we don't delete data like workspaces and prebuilds which we could consider doing in the future, we do remove project-level environments variables, right? See relevant design from https://github.com/gitpod-io/gitpod/issues/7273#issuecomment-1096680115:

We could :a: open a follow-up issue to discuss data deletion and :b: rephrase this to something like the following:

```suggestion
                    This will delete the project and all project-level environment variables you've set for this project.
```

File: components/dashboard/src/components/ConfirmationModal.tsx [hunk 3]
```diff
@@ -20,7 +20,7 @@ export default function ConfirmationModal(props: {
     onConfirm: () => void;
 }) {
     const children: React.ReactChild[] = [
-        <p key="areYouSure" className="mt-3 mb-3 text-base text-gray-500">
+        <p key="areYouSure" className="mb-3 text-base text-gray-500">
```
Comment (gtsiolis) [inline components/dashboard/src/components/ConfirmationModal.tsx:23]: **praise**: Thanks for making this spacing change! Cross-posting relevant comment from https://github.com/gitpod-io/gitpod/pull/15316#discussion_r1046881412 for more context.

File: components/dashboard/src/projects/Projects.tsx [hunk 4]
```diff
@@ -220,96 +181,14 @@ export default function () {
...
-                                        ) : (
-                                            <div className="flex h-full text-md">
-                                                <p className="my-auto ">No recent prebuilds</p>
-                                            </div>
-                                        )}
-                                    </div>
-                                </div>
-                            ))}
+                        {sortedFilteredProjects.map((p) => (
+                            <ProjectListItem
```
Comment (gtsiolis) [inline components/dashboard/src/projects/Projects.tsx:185]: **praise**: Nice!

**Review by gtsiolis — COMMENTED** [review]
Thanks @selfcontained for making these changes and @svenefftinge for the ping[[1](https://github.com/gitpod-io/gitpod/pull/15318#discussion_r1048111111)]! 🏓

File: components/dashboard/src/projects/ProjectSettings.tsx (hunk 2 above)
Comment (svenefftinge) [reply components/dashboard/src/projects/ProjectSettings.tsx:264]: Ah! That ('remove') makes sense, thanks for clarifying 🙏

File: components/dashboard/src/projects/ProjectSettings.tsx (hunk 1 above)
Comment (selfcontained) [reply components/dashboard/src/projects/ProjectSettings.tsx:266]: Ah my mistake, I copied this wording from another related issue. I'll adjust this specific text back to how it was. Thanks for catching.

Comment (selfcontained) [reply components/dashboard/src/projects/ProjectSettings.tsx:266]: Would it be worthwhile to keep this a bit more generic and future-proof, something like:

> This will delete the project and all project-level data you've set for this project.

... or is that too vague and better that we're explicit? I assume we'd have to revisit this wording if we adjust what we delete related to a project being removed over time.

Comment (selfcontained) [reply components/dashboard/src/projects/ProjectSettings.tsx]: Going with your suggestion to just be more clear for now with what happens.
